core/rawdb/freezer.go

```diff
@@ -342,8 +342,16 @@ func (f *Freezer) TruncateHead(items uint64) (uint64, error) {
 	if oitems <= items {
 		return oitems, nil
 	}
-	for _, table := range f.tables {
-		if err := table.truncateHead(items - f.offset); err != nil {
+	for kind, table := range f.tables {
+		err := table.truncateHead(items - f.offset)
+		// if truncate below tail, try reset to it
+		if err == errTruncationBelowTail {
+			nt, err := table.resetItems(items-f.offset, items-f.offset)
```

Comment:
This resetting was the only place with the bug or was there any other place as well?

Reply:
It will be triggered when the chain rewinds to genesis, it will truncate the head till genesis, but the blob table is special, it has the same head, but a different tail from other tables (like bodies, receipts).

So if the chain is rewound to head below the blob's tail, it needs to reset again.